### ISSUE TITLE
ActiveModel::MissingAttributeError: can't write unknown attribute

### ISSUE THREAD

**Issue by danieljacobarcher** [composite-primary-keys/composite_primary_keys#240]
I have two models that share a join table between them, and on the join table model, I used:
````ruby
self.primary_keys = :property_id, :user_id
````

unfortunately, it's giving me a problem when trying to write attributes for every other model. Any one else have these problems or have suggestions?

Thanks!

**Comment by Vanuan** [comment]
Have this too. The change was somewhere in between `x.0.7` and `x.0.13`:

```
ActiveModel::MissingAttributeError: missing attribute: foo

from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/attribute_methods/read.rb:22:in `block (2 levels) in read_attribute'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/attribute_methods/read.rb:21:in `fetch'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/attribute_methods/read.rb:21:in `block in read_attribute'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/attribute_methods/read.rb:11:in `fetch'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/attribute_methods/read.rb:11:in `read_attribute'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/activerecord-4.1.10/lib/active_record/attribute_methods.rb:383:in `[]'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/associations/preloader/association.rb:68:in `block (2 levels) in owners_by_key'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/associations/preloader/association.rb:67:in `map'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/associations/preloader/association.rb:67:in `block in owners_by_key'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/associations/preloader/association.rb:64:in `each'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/associations/preloader/association.rb:64:in `group_by'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/associations/preloader/association.rb:64:in `owners_by_key'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/composite_primary_keys-7.0.13/lib/composite_primary_keys/associations/preloader/association.rb:19:in `associated_records_by_owner'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/activerecord-4.1.10/lib/active_record/associations/preloader/singular_association.rb:9:in `preload'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/activerecord-4.1.10/lib/active_record/associations/preloader/association.rb:20:in `run'
from /Users/john/.rvm/gems/ruby-2.1.2/gems/activerecord-4.1.10/lib/active_record/associations/preloader.rb:137:in `block (2 levels) in preloaders_for_one'
```

x.0.7 returns nil, x.0.13 raises an exception.

https://github.com/composite-primary-keys/composite_primary_keys/blob/v7.0.7/lib/composite_primary_keys/attribute_methods/read.rb

https://github.com/composite-primary-keys/composite_primary_keys/blob/v7.0.12/lib/composite_primary_keys/attribute_methods/read.rb

**Comment by Vanuan** [comment]
Here it is:

https://github.com/composite-primary-keys/composite_primary_keys/commit/0cc608cde988ad0bc37beebe5252e67a28de91a3

**Comment by cfis** [comment]
Going to close this since its so old.

**Comment by erlingur** [comment]
I just bumped into this problem now. Running 10.0.2 with Rails 5.1.4

**Comment by cfis** [comment]
Do you have a traceback?

**Comment by erlingur** [comment]
So I figured out what it was and it was something else. I saw composite_primary_keys in the traceback and figured it had to have something to do with it because I had verified everything else (I thought). After ripping composite_primary_keys out it still happened and I was almost losing my mind after trying everything. Finally figured out that it was because my test DB had not been updated to include the attributes but the dev db had. I ran the required migrations on the test db and everything worked fine. Sorry to waste your time!
